src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -139,6 +140,7 @@ func NewStatefulSet(
 		decommissionWaitInterval,
 		logger.WithValues("Kind", statefulSetKind()),
 		nil,
+		0,
```

Comment:
It seems there's no way to customize this. Is this expected?

Reply:
Yes and no. In ideal world Redpanda operator should have close loop for rolling update. When it detects the need for restart, then Redpanda should be throttled to the point where operator can observe no lag on all Redpanda instances. This is ideal, but in real deployment we can not throttle producers, so there are to options:
1. threshold in cluster custom resource definition
2. threshold in operator configuration (currently command line flags)

I would lean towards threshold in cluster custom resource definition.

What are your thoughts?

Reply:
Yeah I would also feel better if this is customizable - in case this yields cluster never catching up so we can at least tweak it and let it move on... 

I am fine with both options you presented.

Reply:
There's a restartConfig section in the CRD where this can fit. I'm thinking to cases where we do an upgrade between slightly incompatible versions and the cluster is unable to replicate partitions until all nodes are at the same version. The flag allows to unlock the upgrade..